### PR TITLE
Argue graph weighting and dashboard

### PR DESCRIPTION
**Pull request by lfaucon**
builds on top of #621 and #633

### CODE REVIEW

**Comment by houshuang** [comment]
Is this updated with your latest work that you showed yesterday? Could you also put the graph into frog-graphs for testing? Thanks

**Comment by lfaucon** [comment]
@houshuang This should be good to merge now. Sorry it changes a lot of things and you will have trouble reading the code as I refactored everything.

Improvements:
- The updates to the reactiveData happens now only one question at a time
- Fix #744 
- Add argue weighting in config and dashboard to show where are the students.

**Comment by houshuang** [comment]
@lfaucon Hi, I merged develop, and all the examples in preview seem to work well, but I don't see any of the argue graph stuff - ah, if I configure it with weights, it does display. Could you add an exampleData with weights?

Why does it say y 1,3 instead of Chen LI 1, 3?
screen shot 2018-01-18 at 11 14 01

So this only works with weights, not with quiz distance? Or I guess it works with both - that logic is all in the operator? 

There is a small bug where whether an answer is correct or not is always displayed, even though you uncheck that button. This might have been there already.

The dashboard visualisation is cool. Too bad I can't imagine many instructors doing this manually, but... 

Anything else I should test?

**Comment by lfaucon** [comment]
- There is an example with weight called "Opinion Quiz". Maybe you did not see it because of a small bug concerning the dashboard in preview, when you change example you should close and open the dashboard to refresh it
- Having a 2D projection without the XY weights need some ML and will not really work with a small number of students. For example with two students the projection on the plane would for example either put both of them at (0,0) if they are identical of both of them at (1,1) and (-1,-1) if they have even the slightest difference
- I actually never see the display of correct answer. Where does that appear? The PR is quite old so some code can have been lost in a merge of develop.
- You can look at the data exports as I changed a bit the format to be sure you are ok with it

File: frog/imports/ui/Preview/Preview.js [hunk 1]
```diff
@@ -321,8 +321,8 @@ export const StatelessPreview = withState('reload', 'setReload', '')(
                     reload={reload}
                     doc={dashboard}
                     instances={users
-                      .filter((y, i) => i % 2 !== 0)
-                      .map((z, i) => Math.ceil(i / 2))}
+                      .filter((_, i) => i % 2 !== 0)
+                      .map((_, i) => i + 1)}
```
Comment (houshuang) [inline frog/imports/ui/Preview/Preview.js:325]: Why this change? I see removing y and z, is there any difference in the result of Math.ceil vs i+1?

**Comment by houshuang** [comment]
@lfaucon 
Almost ready to merge, but I'm getting ShareDB errors in Preview. Just clicking on an item gives me

```js
Uncaught Error: invalid / missing instruction in op
    at Object.json.apply (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:67755)
    at Doc._otApply (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:63675)
    at Doc._submit (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:63773)
    at Doc.submitOp (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:63871)
    at Doc.objInsert (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:103188)
    at Object.mergeLog (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:309139)
    at mergeData (Preview.js:132)
    at logger (createLogger.js:28)
    at Object.onChange (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:308808)
    at modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:140445
    at run (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:145446)
    at runIfPresent (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:145475)
    at onGlobalMessage (modules.js?hash=e495edb7f8759efc9e7bbec01d914b633307efb1:145515)
    at d (raven.js:386)
```

When I try to run an actual graph, it crashes the server, and disconnects the client...

```js
W20180123-10:43:19.079(1)? (STDERR) events.js:183
W20180123-10:43:19.079(1)? (STDERR)       throw er; // Unhandled 'error' event
W20180123-10:43:19.080(1)? (STDERR)       ^
W20180123-10:43:19.080(1)? (STDERR)
W20180123-10:43:19.080(1)? (STDERR) Error: invalid / missing instruction in op
W20180123-10:43:19.081(1)? (STDERR)     at Object.json.apply (/Users/stian/src/frog/node_modules/ot-json0/lib/json0.js:230:13)
W20180123-10:43:19.081(1)? (STDERR)     at Doc._otApply (/Users/stian/src/frog/node_modules/sharedb/lib/client/doc.js:552:27)
W20180123-10:43:19.081(1)? (STDERR)     at Doc._submit (/Users/stian/src/frog/node_modules/sharedb/lib/client/doc.js:650:8)
W20180123-10:43:19.082(1)? (STDERR)     at Doc.submitOp (/Users/stian/src/frog/node_modules/sharedb/lib/client/doc.js:748:8)
W20180123-10:43:19.082(1)? (STDERR)     at Doc.objInsert (/Users/stian/src/frog/frog-utils/dist/generateReactiveFn.js:103:16)
W20180123-10:43:19.082(1)? (STDERR)     at Object.mergeLog (/Users/stian/src/frog/ac/ac-quiz/dist/Dashboard.js:71:10)
W20180123-10:43:19.082(1)? (STDERR)     at prepareDoc (server/mergeLogData.js:31:30)
W20180123-10:43:19.083(1)? (STDERR)     at Doc.<anonymous> (server/mergeLogData.js:41:17)
W20180123-10:43:19.083(1)? (STDERR)     at Object.onceWrapper (events.js:313:30)
W20180123-10:43:19.083(1)? (STDERR)     at emitNone (events.js:106:13)
W20180123-10:43:19.083(1)? (STDERR)     at Doc.emit (events.js:208:7)
W20180123-10:43:19.084(1)? (STDERR)     at Doc.ingestSnapshot (/Users/stian/src/frog/node_modules/sharedb/lib/client/doc.js:194:8)
W20180123-10:43:19.085(1)? (STDERR)     at Doc._handleFetch (/Users/stian/src/frog/node_modules/sharedb/lib/client/doc.js:243:8)
W20180123-10:43:19.085(1)? (STDERR)     at Connection.handleMessage (/Users/stian/src/frog/node_modules/sharedb/lib/client/connection.js:223:20)
W20180123-10:43:19.085(1)? (STDERR)     at StreamSocket.socket.onmessage (/Users/stian/src/frog/node_modules/sharedb/lib/client/connection.js:117:18)
W20180123-10:43:19.086(1)? (STDERR)     at /Users/stian/src/frog/node_modules/sharedb/lib/stream-socket.js:59:12
```

**Comment by lfaucon** [comment]
Should work now

**Review by houshuang — APPROVED** [review]

**Review by houshuang — APPROVED** [review]
